src/annoylib.h

```diff
@@ -530,9 +535,17 @@ template<typename S, typename T, typename Distance, typename Random>
       Node* m = _get(item);
       m->n_descendants = (S)indices.size();
 
-      // Using std::copy instead of a loop seems to resolve issues #3 and #13,
-      // probably because gcc 4.8 goes overboard with optimizations.
+    // Using std::copy instead of a loop seems to resolve issues #3 and #13,
+    // probably because gcc 4.8 goes overboard with optimizations.
+
+    // (but copy not cooperating with MSVC in certain conditions)
+#if (defined(_MSC_VER) && defined(_DEBUG)) || (defined(_MSC_VER) && (_MSC_VER == 1500))
```

Comment:
IIRC there's actually two places that uses `std::copy` in the code.

might be worth making this a macro btw so that we don't clutter the code with too much compiler specific stuff